src/mail/view/MailViewerViewModel.ts

```diff
@@ -1043,6 +1042,19 @@ export class MailViewerViewModel {
 		}
 	}
 
+	async downloadAndParseVCard(file: TutanotaFile) {
+		const { vCardFileToVCards } = await import("../../contacts/VCardImporter.js")
+		file = (await this.cryptoFacade.enforceSessionKeyUpdateIfNeeded(this._mail, [file]))[0]
+		try {
+			const fileData = await this.fileController.getFileData(file, file.mimeType ?? VCARD_MIME_TYPES.X_VCARD)
+			const decoder = new TextDecoder("utf-8")
+
+			return vCardFileToVCards(decoder.decode(fileData))
+		} catch (e) {
+			await Dialog.message("errorDuringFileOpen_msg")
```

Comment:
dialogs don't belong here, we can throw a `UserError` though or return some kind of result